fabric-mining-level-api-v1/src/main/java/net/fabricmc/fabric/api/mininglevel/v1/FabricTool.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.mininglevel.v1;
+
+import org.jetbrains.annotations.Nullable;
+
+import net.minecraft.block.Block;
+import net.minecraft.block.BlockState;
+import net.minecraft.item.ToolMaterial;
+import net.minecraft.tag.Tag;
+
+/**
+ * Interface for tools to provide information about their mining ability.
+ * Allows items that don't extend {@code MiningToolItem}, such as Swords and Shears, to provide mining information.
+ */
+public interface FabricTool {
+	/**
+	 * Returns the {@code ToolMaterial} of this tool.
+	 * Can be {@code null} if the tool does not utilise {@code ToolMaterial}s.
+	 * Defaults to {@code null}.
+	 * @return The tool material
+	 */
+	@Nullable
+	default ToolMaterial getToolMaterial() {
+		return null;
+	}
+
+	/**
+	 * Returns a {@code Tag} of the blocks this tool is effective against.
+	 * Can be {@code null} if the tool does not utilise a tag.
+	 * Defaults to {@code null}.
+	 * @return The effective blocks
+	 */
+	@Nullable
+	default Tag<Block> getEffectiveBlocks() {
```

Comment:
What about tools that can work on multiple kinds of blocks, e.g. paxels?

Reply:
These can either return a tag that nests `minecraft:mineable/pickaxe`, `minecraft:mineable/axe` and `minecraft:mineable/shovel`, or return null as per the javadoc.

Reply:
I'm not sure I see the point of these functions then.

Reply:
`getToolMaterial` exists so that things like shears can be tied to a tool material, and thus inherit their mining level from it
`getEffectiveBlocks` exists so `isSuitableFor` and `getMiningSpeedMultiplier` can use them
What is so hard to understand about that?

`FabricTool` isn't intended for making fancy multitools or anything, but simply allowing for the creation of simple vanilla-style tools that aren't pickaxes, axes, shovels, or hoes. This API specifically uses it for swords and shears, but it can potentially be used for other things too.

`DynamicAttributeTool` extends `FabricTool`, and it is then ideal for use for making more fancy tools like paxels and drills and whatnot.

Reply:
Imo what these methods are being used for is not super clear. Is it just metadata? Or does it have some special behavior? What is the effect of overriding them? Perhaps they could be extended a tiny little bit to replace the dynamic tool module entirely?